.github/workflows/template.yml

```diff
@@ -58,10 +58,20 @@ jobs:
         with:
           r-version: 'release'
 
-      - name: Install R packages
+      - name: Install neeeded packages
         if: steps.check-rmd.outputs.count != 0
         run: |
-          install.packages('remotes', 'rprojroot', 'renv')
+          install.packages(c('remotes', 'rprojroot', 'renv', 'desc'))
+        shell: Rscript {0}
+
+      - name: Query dependencies
+        if: steps.check-rmd.outputs.count != 0
+        working-directory: lesson
+        run: |
+          source('bin/dependencies.R')
+          deps <- identify_dependencies()
+          create_description(deps)
+          writeLines(sprintf("R-%i.%i", getRversion()$major, getRversion()$minor), ".github/R-version")
         shell: Rscript {0}
 
       - name: Cache R packages
```

Comment:
`Cache R packages` step has to precede the step whose packages it caches and the step that we'd like to skip has to have `if: steps.cache-r.outputs.cache-hit != 'true'` as an additional condition for running.

Reply:
I had forgotten an important line :disappointed: but I use the same steps and the same order as the ones that RStudio has developed <https://github.com/r-lib/actions/blob/master/examples/check-standard.yaml>

1. `Install needed packages`: We need the `remotes`, `rprojroot`, `renv` and `desc` packages to check the dependencies, so we install them every time regardless of caching.
2. `Query dependencies`: We look for all the packages that are specifically needed by the lessons, and create an object that contains the names of the packages that are needed, the installed version number (if it's already installed) available from the local library, and the latest version number available from the repository. 
3. `Cache R packages`: The restore key comparison is only based on the OS and the R-version, this means that each run will always restore the cache from the previous run (as long as R hasn't been upgraded). The cached directory is the entire library (the folder where the R packages get installed).
4. `Install system dependencies for R packages` relatively self-explanatory
5. `Install lesson package dependencies`: runs `install_deps` which (a) installs the packages that are missing from the library and (b) update packages that are out-of-date. If we modify the `restore-keys` pattern to include the hash of the `.github/depends.Rds` file (and skip the install step when there is a match), then we would have to reinstall the entire library even when a single package needs an update.

At the end of the workflow, the R package library gets uploaded to the cache.

Reply:
Thanks, François! Makes sense. It's true that we don't always have to use `cache-hit` check -- we can cache stuff and let R do its thing.